flow-server/src/test/java/com/vaadin/flow/internal/BeanUtilTest.java

```diff
@@ -42,6 +127,255 @@ public boolean isExistsInAllPlaces() {
+                assertNotNull(descriptor.getReadMethod());
+                assertNotNull(descriptor.getWriteMethod());
+            } else if ("nested".equals(descriptor.getName())) {
+                foundNested = true;
+                assertEquals(TestNestedBean.class, descriptor.getPropertyType());
+                assertNotNull(descriptor.getReadMethod());
+                assertNotNull(descriptor.getWriteMethod());
+            }
+        }
+        
+        assertTrue("Should find 'name' property", foundName);
+        assertTrue("Should find 'age' property", foundAge);
+        assertTrue("Should find 'nested' property", foundNested);
+    }
+
+    // Test for getBeanPropertyDescriptors with record
+    @Test
+    public void getBeanPropertyDescriptors_record() throws IntrospectionException {
+        List<PropertyDescriptor> descriptors = BeanUtil.getBeanPropertyDescriptors(TestRecord.class);
+        
+        assertNotNull(descriptors);
+        assertEquals(2, descriptors.size()); // recordProperty, recordAge
+        
+        boolean foundRecordProperty = false, foundRecordAge = false;
+        for (PropertyDescriptor descriptor : descriptors) {
+            if ("recordProperty".equals(descriptor.getName())) {
+                foundRecordProperty = true;
+                assertEquals(String.class, descriptor.getPropertyType());
+                assertNotNull(descriptor.getReadMethod());
+                assertNull(descriptor.getWriteMethod()); // Records are immutable
+            } else if ("recordAge".equals(descriptor.getName())) {
+                foundRecordAge = true;
+                assertEquals(int.class, descriptor.getPropertyType());
+                assertNotNull(descriptor.getReadMethod());
+                assertNull(descriptor.getWriteMethod()); // Records are immutable
+            }
+        }
+        
+        assertTrue("Should find 'recordProperty' property", foundRecordProperty);
+        assertTrue("Should find 'recordAge' property", foundRecordAge);
+    }
+
+    // Test for getBeanPropertyDescriptors with interface
+    @Test
+    public void getBeanPropertyDescriptors_interface() throws IntrospectionException {
+        List<PropertyDescriptor> descriptors = BeanUtil.getBeanPropertyDescriptors(TestInterface.class);
+        
+        assertNotNull(descriptors);
+        assertTrue(descriptors.size() >= 1); // interfaceProperty
+        
+        boolean foundInterfaceProperty = false;
+        for (PropertyDescriptor descriptor : descriptors) {
+            if ("interfaceProperty".equals(descriptor.getName())) {
+                foundInterfaceProperty = true;
+                assertEquals(String.class, descriptor.getPropertyType());
+                assertNotNull(descriptor.getReadMethod());
+                assertNotNull(descriptor.getWriteMethod());
+            }
+        }
+        
+        assertTrue("Should find 'interfaceProperty' property", foundInterfaceProperty);
```

Comment:
As there is only one descriptor we should assert directly on that.

```suggestion
        assertEquals("interfaceProperty", descriptors[0].getName());
        assertEquals(String.class, descriptors[0].getPropertyType());
        assertNotNull(descriptors[0].getReadMethod());
        assertNotNull(descriptors[0].getWriteMethod());
```